lib/matplotlib/__init__.py

```diff
@@ -761,30 +818,147 @@ def __getitem__(self, key):
 
         return self._get(key)
 
+    def _get_default(self, key):
+        keys, depth = self._split_key(key)
+        if depth == 1:
+            if key in self.single_key_set:
+                return self._namespace_maps["default"].get(key)
+            # Uncomment the following line and remove the raise statement
+            # to enable getting namespace parameters.
+            # return self._namespace_maps[key]
+            else:
+                raise KeyError(
+                    f"{key} is not a valid rc parameter (see rcParams.keys() for "
+                    f"a list of valid parameters)")
+        elif depth == 2:
+            return self._namespace_maps[keys[0]].maps[-1].get(keys[1])
+
+    def getdefault(self, key):
```

Comment:
We should follow the naming convention here. While some stdlib API does not use underscores for historic reasons (e.g. `dict.setdefault`) we should follow the naming convention and not some special historic cases.
```suggestion
    def get_default(self, key):
```

Reply:
I don't mind changing this but 1. Do we change `setdefault` to `set_default` too? and as you mentioned 2. Do we still keep setdefault?